src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsClientModelValidatorProvider.cs

```diff
@@ -5,26 +5,32 @@
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using System.Linq;
+using Microsoft.AspNet.Mvc.DataAnnotations;
 using Microsoft.Extensions.Localization;
 using Microsoft.Extensions.OptionsModel;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
 {
     /// <summary>
+    /// A factory for validators based on ValidationAttribute.
+    /// </summary>
+    /// <param name="attribute"></param>
+    /// <param name="stringLocalizer"></param>
+    /// <returns></returns>
+    public delegate IAttributeAdapter ValidationAttributeAdapterFactory(
```

Comment:
Not sure about our guidance on delegates sharing files. @dougbu would know better.